File: src/type/application.py
from typing import Optional, TypedDict
from enum import Enum


class Types(Enum):
    APPIMAGE = 'AppImage'
    FLATHUB = 'Flathub'
    SNAP = 'Snapcraft'


class Application(TypedDict):
    name: str
    src: str
    version: Optional[str]
    type: Types
    repo_url: Optional[str]

File: types/application.go
package types

type Type string

const (
	AppImageType Type = "AppImage"
	FlatpakType  Type = "Flathub"
	SnapType     Type = "Snapcraft"
)

type BaseApplication struct {
	Name    string ""
	Url     string ""
	Type    Type
	RepoUrl string
}
